client/www/lib/hooks/useSavedQueryState.tsx

```diff
@@ -0,0 +1,42 @@
+import { debounce } from 'lodash';
+import { useQueryState, UseQueryStateOptions, UseQueryStateReturn } from 'nuqs';
+import { useCallback, useEffect } from 'react';
+
+export const useSavedQueryState = <
```

Comment:
This isn't working for me in the vercel preview. Steps to replicate: 

1. Go to sandbox, enter in "hello"
2. Click on the Explorer tab
3. Click on the sandbox tab, code gets replaced by default value.

I think useQueryState may do something weird if you call setValue before it has had time to initialze? If you get the item from localStorage and set the defaultValue in `useQueryState` as `localStorageValue || defaultValue`, then it seems to work.

Reply:
interesting, i didn't set the defaultvalue there because if the current value matches the default then it doesn't add a param at all, but i realized i can turn that off so ill just do that